src/components/diagrams/diagram-pane.js

```diff
@@ -286,95 +362,314 @@ export function DiagramPane({
         useDiagram();
     const currentNodeRef = useRef();
     currentNodeRef.current = currentNode;
+    const viewsRef = useRef([]);
+    viewsRef.current = views;
     const classes = useStyles();
-    const [warnings, setWarnings] = useState(new Map());
 
     /**
      * BUILDS THE DIAGRAMS LIST
      */
 
-    // Here, the goal is to build a list of view, each view corresponding to a diagram.
+    // Here (for the next 300 lines), the goal is to build a list of view, each view corresponding to a diagram.
     // We get the diagram data from the redux store.
     // In the case of SLD, each SLD corresponds to one view, but in the case of NAD, each open NAD is merged
-    // into one view.
-    useEffect(() => {
-        if (
-            !visible ||
-            isNodeInNotificationList(currentNode, notificationIdList)
-        ) {
-            return;
-        }
-        const diagramViews = [];
-        const networkAreaIds = [];
-        let networkAreaViewState = ViewState.OPENED;
+    // into one view.s
 
-        diagramStates.forEach((diagramState) => {
-            if (diagramState.svgType === DiagramType.NETWORK_AREA_DIAGRAM) {
-                networkAreaIds.push(diagramState.id);
-                networkAreaViewState = diagramState.state; // They should all be the same value
-            } else {
-                let singleLineDiagramView = createView(diagramState);
-                // if current view cannot be found, it returns undefined
-                // in this case, we keep it in the diagram states and show a warning message inside the SLD
-                if (singleLineDiagramView) {
-                    diagramViews.push({
-                        ...singleLineDiagramView,
-                        align: 'left',
-                    });
-                } else {
-                    // because current view cannot be found, the id will be assigned to the name.
-                    const emptyDiagramView = {
-                        ...diagramState,
-                        name: diagramState?.id,
-                        // this variable is used to show a warning message inside the SLD
-                        warningMessage:
-                            DiagramType.VOLTAGE_LEVEL === diagramState.svgType
-                                ? 'VoltageLevelNotFound'
-                                : 'SubstationNotFound',
-                    };
+    // Check if we need to add new SLDs in the 'views' and add them if necessary
+    const addSLDs = useCallback(
```

Comment:
Maybe rename this function to "addMissingSLDs" ?

Reply:
Ok indeed clearer